### PR TITLE
Add new feature:  hijacking and dns cheat

### PR DESCRIPTION
**Pull request by arcanique**
…w feature : dns cheat

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind feature

**What this PR does / why we need it**:
>
>Add hijacking script tools and initContainer image Dockerfile. And dns cheat. and enabled edgemesh module.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: edgemesh/pkg/server/dns.go [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+	idxadd := unsafe.Sizeof(h.additNum) + idxaut
+
+	binary.BigEndian.PutUint16(rspHead[: idxTran], h.transactionID)
+	binary.BigEndian.PutUint16(rspHead[idxTran: idxflag], h.flags)
+	binary.BigEndian.PutUint16(rspHead[idxflag: idxque], h.queNum)
+	binary.BigEndian.PutUint16(rspHead[idxque: idxans], h.ansNum)
+	binary.BigEndian.PutUint16(rspHead[idxans: idxaut], h.authNum)
+	binary.BigEndian.PutUint16(rspHead[idxaut: idxadd], h.additNum)
+	return
+}
```
Comment (kevin-wangzefeng) [inline edgemesh/pkg/server/dns.go]: no new line at the end of file.

File: edgemesh/tools/initContainer/deb/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FROM ubuntu
+
+ADD edgemesh-iptables.sh /usr/local/bin
+RUN apt-get update && apt-get install -y iproute2 iptables
+
+ENTRYPOINT ["usr/local/bin/edgemesh-iptables.sh"]
```
Comment (kevin-wangzefeng) [inline edgemesh/tools/initContainer/deb/Dockerfile]: no new line at the end of file.

File: edgemesh/tools/initContainer/rpm/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FROM centos:latest
+
+ADD edgemesh-iptables.sh /usr/local/bin
+
+RUN yum -y update && yum install -y iproute iptables
+		
+ENTRYPOINT ["/usr/local/bin/edgemesh-iptables.sh"]
```
Comment (kevin-wangzefeng) [inline edgemesh/tools/initContainer/rpm/Dockerfile]: no new line at the end of file.

File: edgemesh/pkg/server/dns.go [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+	twoByteSize = uint16(2)
+	ttl = uint32(64)
+	fakeIp = []byte{5,5,5,5}
+)
+
+const (
+	aRecord = 1
+	bufSize = 1024
+	notImplem = uint16(0x0004)
+
```
Comment (sids-b) [inline edgemesh/pkg/server/dns.go]: please remove blank line

Comment (arcanique) [reply edgemesh/pkg/server/dns.go]: I have modified this problem

File: edgemesh/tools/initContainer/rpm/Dockerfile (hunk 3 above)
Comment (arcanique) [reply edgemesh/tools/initContainer/rpm/Dockerfile]: I have modified this problem

File: edgemesh/tools/initContainer/deb/Dockerfile (hunk 2 above)
Comment (arcanique) [reply edgemesh/tools/initContainer/deb/Dockerfile]: I have modified this problem

File: edgemesh/pkg/server/dns.go (hunk 1 above)
Comment (arcanique) [reply edgemesh/pkg/server/dns.go]: I have modified this problem

File: edgemesh/pkg/server/dns.go [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+		}
+	}
+	return rsp
+}
+
+// parseNameServer gets the nameserver from the resolv.conf
+func parseNameServer()([]net.IP){
+	file, err := os.Open("./etc/resolv.conf")
+	if err != nil {
+		glog.Errorf("the resolv.conf has sone error : %s")
```
Comment (rohitsardesai83) [inline edgemesh/pkg/server/dns.go]: "the resolv.conf has sone error : %s"  -> glog.Errorf("error opening /etc/resolv.conf : %v, err)

File: edgemesh/pkg/server/dns.go [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+	"net"
+	"os"
+	"strings"
+	"unsafe"
+
+	"github.com/golang/glog"
+)
+
+var (
+	inter = "docker0"
```
[the remark this answers is not in the document]
Comment (arcanique) [reply edgemesh/pkg/server/dns.go]: Here is just an interface name. The first version temporarily uses docker0. After the subsequent container network is determined, the corresponding modification is made. I made a record here.

File: edgemesh/pkg/server/dns.go [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+
+	offset := uint16(unsafe.Sizeof(dnsHeader{}))
+	question.getQuestion(req, offset, head)
+
+	que = question
+	err = nil
+	return
+}
+
+//isAQurey judge if the dns pkg is a Qurey process
```
[the remark this answers is not in the document]
Comment (arcanique) [reply edgemesh/pkg/server/dns.go]: sorry about this mistake. I will modify this

File: edgemesh/tools/initContainer/script/edgemesh-iptables.sh [hunk 8]
```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/bash
+
+function usage() {
+	echo 'this is the edgemesh-iptables usage'
+	echo "${0} -p PROXY_PORT [-i HIJACK_IP] [-t HIJACK_PORT] [-b EXCLUDE_IP] [-c EXCLUDE_PORT] [-h]"
+	echo ''
+	echo '  -p: Specify the edgemesh port to which redirct all TCP traffic from the Pod. (default 10001)'
```
[the remark this answers is not in the document]
Comment (arcanique) [reply edgemesh/tools/initContainer/script/edgemesh-iptables.sh]: k

File: edgemesh/tools/initContainer/createImg.sh [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+chmod 0777 edgemesh-iptables.sh
+
+if command -v docker > /dev/null 2>&1 ; then
+	#docker build
+	docker build -t edgemesh_init .
+	# delete iptables script
+	rm ./edgemesh-iptables.sh 
+else
+	exit 1
```
[the remark this answers is not in the document]
Comment (arcanique) [reply edgemesh/tools/initContainer/createImg.sh]: ok

File: edgemesh/tools/initContainer/script/edgemesh-iptables.sh [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+	echo "Container Network mode is: ${NETMODE}"
+	echo "Variables:"
+	echo "EDGEMESH_PROXY_PORT=${EDGEMESH_PROXY_PORT-10001}"
+	echo "EDGEMESH_HIJACK_IP=${EDGEMESH_HIJACK_IP-"*"}"
+	echo "EDGEMESH_HIJACK_PORT=${EDGEMESH_HIJACK_PORT-"*"}"
+	echo "EDGEMESH_EXCLUDE_IP=${EDGEMESH_EXCLUDE_IP-}"
+	echo "EDGEMESH_EXCLUDE_PORT=${EDGEMESH_EXCLUDE_PORT-}"
+	
+	#set -o -e # exit if command return 
+	#set -o -u # kuo zhan yige weishezhi de bianliang shi
```
[the remark this answers is not in the document]
Comment (arcanique) [reply edgemesh/tools/initContainer/script/edgemesh-iptables.sh]: this is some debug info. I will del this line

File: edgemesh/pkg/server/dns.go (hunk 5 above)
Comment (arcanique) [reply edgemesh/pkg/server/dns.go]: k

**Pull request merged**